include/boost/url/grammar/unsigned_decimal.hpp

```diff
@@ -50,6 +49,20 @@ struct unsigned_decimal
     }
 
 private:
+    BOOST_STATIC_ASSERT(
+        std::numeric_limits<
+            Unsigned>::is_integer &&
+        ! std::numeric_limits<
+            Unsigned>::is_signed);
+
+    static constexpr Unsigned Max = (
+        std::numeric_limits<
+            Unsigned>::max)();
+
+    static constexpr Unsigned Digits10 =
```

Comment:
these constants could stay in the function, which is arguable better since if we move the definition to the impl/xxx.hpp eventually, then there is less material in this header.